lib/parser.ex

```diff
@@ -0,0 +1,50 @@
+defmodule DateTimeParser.Parser do
+  @moduledoc false
+  alias DateTimeParser.{Epoch, Serial}
+
+  @epoch_regex ~r|\A(?<seconds>\d{10,11})(?:\.(?<subseconds>\d{1,10}))?\z|
+  @serial_regex ~r|\A-?\d{1,5}(?:\.\d{1,10})?\z|
+  @default_parsers [:epoch, :serial, :tokenizer]
+
+  def get_parser(string, opts) do
+    opts = Keyword.merge([tokenizer: get_tokenizer(opts[:context], string)], opts)
+
+    opts
+    |> Keyword.get(:parsers, Application.get_env(DateTimeParser, :parsers, @default_parsers))
+    |> Enum.find_value({:error, :no_parser}, fn parser ->
+      apply(__MODULE__, parser, [string, opts])
```

Comment:
I trust this does the right thing, but wanted to try to write out what I _think_ it's donig for my own understanding - for each `parser` in our options list,  `apply` essentially allows you to dynamically return a reference to the function we want to call (which is the module's function that matches the parser's name). This allows the [retruned variable/function to be invoked later](https://github.com/taxjar/date_time_parser/pull/22/files#diff-355f2f8e435bca7d2d98fb4c4908095bR215) and call the correct parser function.

Reply:
yep!
This also implies that the user can configure the order of parsers tried. If they wanted to try the tokenizer first, they could and _then_ fallback to epoch or serial parsers.